File: libs/java/grpc-client/src/main/java/arda/grpc/interceptor/ClientLoggingInterceptor.java
package arda.grpc.interceptor;

import io.grpc.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ClientLoggingInterceptor implements ClientInterceptor {
    private static final Logger logger = LoggerFactory.getLogger(ClientLoggingInterceptor.class);

    @Override
    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
            MethodDescriptor<ReqT, RespT> method,
            CallOptions callOptions,
            Channel next) {
        logger.debug("gRPC Call: {}", method.getFullMethodName());
        return next.newCall(method, callOptions);
    }
}

File: arda-be-java/libs/grpc-client/src/main/kotlin/arda/grpc/interceptor/ClientLoggingInterceptor.kt
package arda.grpc.interceptor

import io.grpc.*
import org.slf4j.LoggerFactory

class ClientLoggingInterceptor : ClientInterceptor {
    private val logger = LoggerFactory.getLogger(ClientLoggingInterceptor::class.java)

    override fun <ReqT, RespT> interceptCall(
        method: MethodDescriptor<ReqT, RespT>,
        callOptions: CallOptions,
        next: Channel
    ): ClientCall<ReqT, RespT> {
        logger.debug("gRPC Call: ${method.fullMethodName}")
        return next.newCall(method, callOptions)
    }
}
